kb/development/mbedtls-coding-standards.md

```diff
@@ -8,102 +8,101 @@ This document describes Mbed TLS preferences for code formatting, naming convent
+exit:
+    ...cleanup...
+    return ret;
+}
 ```
 
-### Related lines: Multi-line formatting and indentation
+Every code statement should be on its own line, except in `for` loop headers.
 
-Multiple related source code lines should be formatted to be easily readable:
-```c
-#define GET_UINT32_LE( n, b, i )                        \
-{                                                       \
-    (n) = ( (uint32_t) (b)[(i)    ]       )             \
-        | ( (uint32_t) (b)[(i) + 1] <<  8 )             \
-        | ( (uint32_t) (b)[(i) + 2] << 16 )             \
-        | ( (uint32_t) (b)[(i) + 3] << 24 );            \
-}
+### Space placement
 
-if( my_super_var == second_super_var &&
-    this_check_will_do != the_other_value )
+Put a space in the following places:
 
-do_function( ctx, this_is_a_value, value_b,
-                  the_special_var );
+* Around binary operators: `(x + y) * z`
+* After a comma: `f(x, y)`
+* Before the asterisk in a pointer type: `int *p`
+* Between `if`, `switch`, `while`, `for` and the opening parenthesis: `if (f(x))`
+* Outside curly braces: `do { ... } while (!done)`, `struct mystruct s = { 0 }`
+* Inside the curly braces around initializers: `int a[] = { 1, 2, 3 };`
 
-void this_is_a_function( context_struct *ctx, size_t length,
-                         unsigned char *result );
-```
+Do not put a space in the following places:
 
-### Extra parentheses for `return` and `sizeof`
+* After the function or macro name in a function or macro call: `f(x, y)`
+* Inside parentheses: `if (f(sizeof(int), (int) y))`
+* Inside square brackets: `a[i + 1]`
+* Before a comma: `f(x, y)`
+* After a unary operator: `if (!condition)`, `++x`
+* Before a unary operator: `x++`
+* Between an array and the following opening bracket: `int a[2]; a[i]`
+* Around field access symbols: `s.a`, `p->a`
+* After the asterisk in a pointer type: `int *p`
+* Between the asterisks double pointers types or derefences: `char **p`, `x + **p`
 
-Within Mbed TLS return statements use parentheses to contain their value:
+### Use of parentheses
+
+`sizeof` expressions always use parentheses even when it is not necessary (when taking the size of an object):
 ```c
-    return( 0 );
+    memset(buf, 0, sizeof(buf));
 ```
-Similarly, sizeof expressions always use parentheses even when it is not necessary (when taking the size of an object):
+
+### Braces placement and block declaration
+
+Braces are mandatory in control statements such as `if`, `while`, `for`, even when the content is a simple statement.
+Opening braces (curly brackets) are on the same line as the control statement.
+
+Closing braces are aligned with the control statement or opening brace. They are generally alone on their line, except when followed by `else` or by the `while` of a `do ... while` statement.
+
 ```c
-    memset( buf, 0, sizeof( buf ) );
+    do {
+        x += f();
+        if (x == 0) {
+            continue;
+        }
+        x += g();
+    } while (condition);
```

Comment:
I think we should state explicitly somewhee that for function definitions, the opening brace is on its own line. This section looks like it's about braces placement in general, but so far its content only mentions one case: control statements.

(Btw, function definitions is no the only missing case: enum/struct/union, and array initializers also come to mind: they're mentioned in "Formatting of lists" below but I don't think anything's said explicitly about the placement of braces.)